docs/design/storage-auto-scale.md

```diff
@@ -58,23 +58,41 @@ Go-routine Scraper:
 1) Add a single go-routine to watch periodically(10mins) all the OSDs sizes using Prometheus metrics in the storagecluster namespace.
     Query: (ceph_osd_metadata * on (ceph_daemon, namespace, managedBy) group_right(device_class,hostname) (ceph_osd_stat_bytes_used / ceph_osd_stat_bytes))
 
-2) With every run calculate different device class highest osd filled percentage.
-   
-        OsdPercentage:
+2) With every run calculate different device class highest osd filled percentage, osdCount and osdSize.
+    
+        OsdInfo:
         
-        ssd               69%
-        nvme              71%
+        ssd{               
+            osdPercentage: 69%
+            osdCount:      3
+            osdSize:       100Gi
+        }
+        nvme{               
+            osdPercentage: 71%
+            osdCount:      3
+            osdSize:       200Gi
+        }
         ...   
 
-3) Create a sync map with `OsdPercentage`, Send a event to the go channel.
+3) Create a sync map with `OsdInfo`, Send a event to the go channel.
 
 Controller:
 
 1) Create a named controller that watches for [channel generic event](https://book-v1.book.kubebuilder.io/beyond_basics/controller_watches) per device class.
 
-2) If the expansion is not in progress set the status `phase` to `NotStarted` 
+2) Set the status `phase` to `NotStarted`, if no expansion has triggered(status.phase==""). 
+   
+3) Check if the expansion is in progress:
+   
+   1) Load the actualOsdCount and actualOsdSize from the syncMap.
+   
+   2) Load the desiredOsdCount and desiredOsdSize from the Storagecluster.
```

Comment:
The desired size is simple to load directly, but the desired osd count needs to be calculated for each storageDeviceSet settings.
```suggestion
   2) Calculate the desiredOsdCount and desiredOsdSize from the Storagecluster.
```